client/src/modules/prices/modal.js

```diff
@@ -14,10 +14,27 @@ function PriceListModalController( $uibModalInstance, Inventory, util, Notify) {
   vm.action = null;
   vm.isValidItem = isValidItem;
 
-  // load Inventory
+  // Here we load the All items from the list
+  const cache = new AppCache('selecedItems');
+  const selectedItems = cache.items;
+
   Inventory.read()
     .then(function (inventory) {
-      vm.inventories = inventory;
+      // Let filter the list,
+      // all items in from the list should not appear in the select dropdown list
+
+      vm.inventories = inventory.map(function (item) {
```

Comment:
This is an interesting idea!

I think what you want to do here is actually an [Array.prototype.map()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) and then an [Array.prototype.filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter).  Here's how you can make this really simple:
```js
// gather all used uuids in a flat array
var selectedUuids = selectedItems.map(function (item) {
  return item.inventory_uuid;
});

// filter the inventory to only show ones that are not used.
vm.inventories = inventory.filter(function (item) {
  return selectedUuids.indexOf(item) === -1;
});
```